sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/NestedDataOperatorConversions.java

```diff
@@ -395,24 +465,182 @@ public DruidExpression toDruidExpression(
+  {
+    private static final SqlFunction FUNCTION =
+        OperatorConversions.operatorBuilder("JSON_VALUE_ANY")
+                           .operandTypeChecker(
+                               OperandTypes.or(
+                                   OperandTypes.sequence(
+                                       "(expr,path)",
+                                       OperandTypes.family(SqlTypeFamily.ANY),
+                                       OperandTypes.family(SqlTypeFamily.STRING)
+                                   ),
+                                   OperandTypes.family(
+                                       SqlTypeFamily.ANY,
+                                       SqlTypeFamily.CHARACTER,
+                                       SqlTypeFamily.ANY,
+                                       SqlTypeFamily.ANY,
+                                       SqlTypeFamily.ANY,
+                                       SqlTypeFamily.ANY,
+                                       SqlTypeFamily.ANY
+                                   )
+                               )
+                           )
+                           .operandTypeInference((callBinding, returnType, operandTypes) -> {
+                             RelDataTypeFactory typeFactory = callBinding.getTypeFactory();
+                             if (operandTypes.length > 5) {
+                               operandTypes[3] = typeFactory.createSqlType(SqlTypeName.ANY);
+                               operandTypes[5] = typeFactory.createSqlType(SqlTypeName.ANY);
+                             }
+                           })
+                           .returnTypeInference(
+                               ReturnTypes.cascade(
+                                   opBinding -> opBinding.getTypeFactory().createTypeWithNullability(
+                                       // STRING is the closest thing we have to an ANY type
+                                       // however, this should really be using SqlTypeName.ANY.. someday
+                                       opBinding.getTypeFactory().createSqlType(SqlTypeName.VARCHAR),
+                                       true
+                                   ),
+                                   SqlTypeTransforms.FORCE_NULLABLE
+                               )
+                           )
+                           .functionCategory(SqlFunctionCategory.SYSTEM)
+                           .build();
 
     @Override
     public SqlOperator calciteOperator()
     {
-      return SqlStdOperatorTable.JSON_VALUE_ANY;
+      return FUNCTION;
+    }
+
+    @Nullable
+    @Override
+    public DruidExpression toDruidExpression(
+        PlannerContext plannerContext,
+        RowSignature rowSignature,
+        RexNode rexNode
+    )
+    {
+      final RexCall call = (RexCall) rexNode;
+
+      // calcite parser can allow for a bunch of junk in here that we don't care about right now, so the call looks
```

Comment:
Some of the extra parameters look like things that are meant to alter the behavior of the function. (like controlling what the error behavior is.) If we don't support them, better to throw an error than silently ignore them. Is this feasible?

Reply:
I couldn't find an easy way, but it is certainly possible that I didn't try everything yet. The reason being afaict the parser creates default values for all of these arguments when it is making the sql function call, and so it always first appears with 6 or 7 arguments and i wasn't too sure how to distinguish them explicitly being there from them being generated.

I can play around a bit to see if I can detect it, or at least maybe expect the default values that are generated and throw an error if they do not exist

Reply:
I think i've addressed everything except for this and finding a way to not have the internal functions directly callable. Since it isn't a new behavior that we ignore these parameters, does this need to block this PR?

Reply:
It doesn't need to block this PR. I'd encourage it as a follow up.

Btw, the "expect the default parameters" thing is what I was imagining. That way, if someone explicitly specifies the default, it still works fine. But if they try to use a different option, they get an error.

Reply:
ended up adding checks in this PR that only the default parameters are allowed (which thankfully matches the behavior we have implemented)